src/spikeinterface/extractors/neoextractors/neuronexus.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+from spikeinterface.core.core_tools import define_function_from_class
+
+from .neobaseextractor import NeoBaseRecordingExtractor, NeoBaseSortingExtractor
+
+
+class NeuroNexusRecordingExtractor(NeoBaseRecordingExtractor):
+    """
+    Class for reading data from NeuroNexus Allego.
+
+    Based on :py:class:`neo.rawio.NeuronexusRawIO`
+
+    Parameters
+    ----------
+    file_path : str
+        The file path to the metadata .xdat.json file of an Allego session
+    stream_id : str | None, default: None
+        If there are several streams, specify the stream id you want to load.
+    stream_name : str | None, default: None
+        If there are several streams, specify the stream name you want to load.
+    all_annotations : bool, default: False
+        Load exhaustively all annotations from neo.
+    use_names_as_ids : bool, default: False
```

Comment:
Can we provide an illustrative examples of the names or are they too variable so giving a pattern does not make sense?

Reply:
I think we don't know the pattern super well. We only have the one test file at Neo, so I could write something, but I'm not confident.

Reply:
Let me know if you think it is still worth putting and I can combine your two pieces of feedback.

Reply:
Na, let's do it when we know more. No point in adding possible confusing message.